llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2374,31 +2375,31 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
   // (sub (or A, B) (and A, B)) --> (xor A, B)
   {
     Value *A, *B;
-    if (match(Op1, m_And(m_Value(A), m_Value(B))) &&
+    if (match(Op1, m_c_And(m_Value(A), m_Value(B))) &&
```

Comment:
This is unnecessary. The m_And captures the first operand into A and the second operand into B. It doesn't do any other checks so it will always match the first operand order it tries. It will never try to check the other order.